surface/jvm/src/main/scala/wvlet/surface/reflect/ObjectBuilder.scala

```diff
@@ -110,50 +113,85 @@ trait StandardBuilder extends GenericBuilder with LogSupport {
     }
   }
 
+  private def getArrayHolder(name: String): ArrayHolder = {
+    // Clear the default value holder if exists
+    holder.get(name) match {
+      case Some(Value(v)) =>
+        // remove the default value
+        holder.remove(name)
+      case _ => // do nothing
+    }
+    holder.getOrElseUpdate(name, ArrayHolder(new ArrayBuffer[Any])).asInstanceOf[ArrayHolder]
+  }
+
   def set(path: Path, value: Any) {
-    if (!path.isEmpty) {
+    if (path.isEmpty) {
+      // do nothing
+    }
+    else {
       val name = path.head.canonicalName
       val p = findParameter(name)
       if (p.isEmpty) {
         error(s"no parameter is found for path $path")
-        return
       }
-
-      trace(s"set path $path : $value")
-
-      if (path.isLeaf) {
-        val valueType = p.get.surface
-        trace(s"update value holder name:$name, valueType:$valueType (isArray:${isArray(valueType)}) with value:$value")
-        if (canBuildFromBuffer(valueType)) {
-          val gt = valueType.typeArgs(0)
-
-          holder.get(name) match {
-            case Some(Value(v)) =>
-              // remove the default value
-              holder.remove(name)
-            case _ => // do nothing
+      else {
+        trace(s"set path $path : $value")
+        if (path.isLeaf) {
+          val targetType = p.get.surface
+          trace(s"update value holder name:$name, valueType:$targetType (isArray:${isArray(targetType)}) with value:$value (${value.getClass})")
+          if (canBuildFromBuffer(targetType)) {
```

Comment:
A reason we need to prepare ArrayHolder here is we will use ObjectBuilder to parse command line arguments. For example, when parsing 'command -a A1 -a A2 ...', we need to call set("a", "A1"), set("a", "A1") separetely when the result type is Config(a:Seq[String]).